ostd/src/arch/x86/cpu/tdx.rs

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use tdx_guest::{handle_virtual_exception, tdcall, TdxTrapFrame};
+
+use crate::cpu::{RawGeneralRegs, UserContext};
+
+pub(crate) fn handle_virtualization_exception(user_context: &mut UserContext) {
+    let ve_info = tdcall::get_veinfo().expect("#VE handler: fail to get VE info\n");
+    let mut generalrags_wrapper = GeneralRegsWrapper(*user_context.general_regs_mut());
+    handle_virtual_exception(&mut generalrags_wrapper, &ve_info);
+    *user_context.general_regs_mut() = generalrags_wrapper.0;
```

Comment:
So converting between the original and the wrapper types involving copying twice. This can be avoided if `GeneralRegsWrapper` takes a mutable reference of `RawGeneralRegs`.

```rust
struct GeneralRegsWrapper<'a>(&'a mut RawGeneralRegs);
```